### ISSUE TITLE
[Bug]: In iOS 17 , can't request Full Access , this is the latest Calendar access levels in iOS 17.

### ISSUE THREAD

**Issue by Allenxuxu** [Baseflow/flutter-permission-handler#1108]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-permission-handler/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-permission-handler/blob/main/permission_handler/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [ ] Android
- [X] iOS
- [ ] Windows

### Steps to reproduce

1. In iOS 17 : https://developer.apple.com/documentation/technotes/tn3152-migrating-to-the-latest-calendar-access-levels 
2. I can't request Full Access

### Expected results

Full Access can be granted by requesting.

### Actual results

Only can request Add Events Only

### Code sample

<details><summary>Code sample</summary>

```dart

var status = await Permission.calendar.request();

```

</details>

### Screenshots or video

<details>
<summary>Screenshots or video demonstration</summary>

[Upload media here]
<img width="417" alt="image" src="https://github.com/Baseflow/flutter-permission-handler/assets/20566897/2559bce8-ad6a-4ae7-a390-a8d72364b8b1">

</details>

### Version

  permission_handler: ^10.4.3

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.1, on macOS 13.4.1 22F82 darwin-x64, locale zh-Hans-CN)
    • Flutter version 3.10.1 on channel stable at /Users/xuxu/flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d3d8effc68 (8 weeks ago), 2023-05-16 17:59:05 -0700
    • Engine revision b4fb11214d
    • Dart version 3.0.1
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/xuxu/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.2.3)
    • IntelliJ at /Users/xuxu/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.2.3)
    • IntelliJ at /Users/xuxu/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/222.4345.14/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 89E998A2-CAB6-4A9B-B4B1-AB81DD5D4347 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-0 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 13.4.1 22F82 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 114.0.5735.198

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by JeroenWeener** [comment]
Thank you for filing this issue @Allenxuxu!

It seems like iOS changed the Calendar permission API in iOS 17. Will label this as an enhancement.

**Comment by parker-sherrill** [comment]
Need this resolved!

**Comment by Pautex** [comment]
I solve full calendar access bug after 3 days of work. My app 'Fetes' needs write and read calendar in order to remove own event, I proceed with two steps, first you granted requestWriteOnlyAccessToEvent..  accept it, wait for granded and after that you post a second request : requestFullAccesToEvent.. and you will be granded full acces, user need reply and accept 2 times but that work well. User can now set and remove access into application settings if any.
Of course you need to add in info.plist file with writeOnlyAccess... AND fullAccess.. items for descriptor about calendar usage. Tthis post is ONLY for iOS17, you need to call such kind of access only with iOS17 and more, proceed as usual for iOS 10...16. A bad or old request in iOS 17 remove granded access.

**Comment by stephenjen** [comment]
Has this been resolved?  I'm tried await Permission.calendar.request() on both iOS 17.0 and 17.1 and am getting PermissionStatus.permanentlyDenied even after changing to a new bundle id.

@Pautex, can you share the workaround you mentioned?  I've tried the following:

`PermissionStatus status = await Permission.calendar.request();
PermissionStatus status1 = await Permission.calendarReadOnly.request();
PermissionStatus status2 = await Permission.calendarFullAccess.request();`

**Comment by JeroenWeener** [comment]
@stephenjen we are close to merging #1151.

**Comment by Pautex** [comment]
Stephenjen don't ask Permission.calendar on iOS17 that ruin effort

**Comment by stephenjen** [comment]
@Pautex taking out Permission.calendar does not resolve the issue for me

**Comment by Quentinio** [comment]
Any news regarding this?

**Comment by JeroenWeener** [comment]
This issue should not be closed yet.

@Quentinio we have updated the individual platforms. We hope to put out a new release today.

**Comment by Quentinio** [comment]
@JeroenWeener I found a work around by invoking method between swift and dart however when using plugin such as device_calendar I need to restart the app for changes to be recognised. Thank you and waiting for update :)

**Comment by Zahid046** [comment]
@Quentinio  mentioned earlier its need to be restarted for changes to be take effect.

**Comment by stephenjen** [comment]
Hmmm, I just restarted my phone, uninstalled and reinstalled the app, and the issue is still there

**Comment by Pautex** [comment]
stephenjen for full calendar access proced into 2 steps.
first ask for write only and accept it.
Second step ask full access when necessary but don't use in iOS17 old method
you must select iOS version < iOS 17 or >= iOS 17

**Comment by Pautex** [comment]
- (void)viewDidLoad {
    [super viewDidLoad];
    isGranted = FALSE;          // defaut invalide
    
    eventStore	= [[EKEventStore alloc] init];
    
    if (@available(iOS 17.0, *)) {      // a partir de iOS17
        [ eventStore requestWriteOnlyAccessToEventsWithCompletion:^(BOOL granted, NSError *error) {
            NSLog(@"requestFullAccessToRemindersWithCompletion granted (bad si 0) = %d Err : %@", granted, error);
            if(granted) {       // c accepte ok retour OK
                isGranted = granted;        // sera true si ok attend passage premier sinon ok ensuite
            }
        }];
    } else {
        // Fallback on earlier versions
        // sur iOS17 ca coince ici !
        [eventStore requestAccessToEntityType:EKEntityTypeEvent completion:^(BOOL granted, NSError *error) {
            NSLog(@"requestAccessToEntityType granted (bad si 0) = %d", granted);
            if(granted) {              // c accepte ok retour OK
                isGranted = granted;  // sera true si ok attend passage premier sinon ok ensuite
            }
        }];
    }
   ...... 
}

Example fist step ask for write only

**Comment by Pautex** [comment]
And after, only after granted for write only is OK you can ask for reading with FullAccess
with iOS 17 never use 'requestAccessToEntityType' call will remove access !

**Comment by stephenjen** [comment]
Yes!  Thank you @Pautex!  I'm on Flutter so I had to add the following to the Podfile:

`'PERMISSION_EVENTS_FULL_ACCESS=1',`

Then call:

`await Permission.calendarWriteOnly.request();`

And then: 

`await Permission.calendarFullAccess.request();`

And now it works

**Comment by chalananupun** [comment]
If platform is iOS => await Permission.calendarWriteOnly.request()
for android => await Permission.calendarFullAccess.request()
